app/src/main/java/com/axehai/weatherscope/data/remote/utils/SafeApiCall.kt

```
@@ -0,0 +1,27 @@
+ * @param T The type of data returned by the [block].
+ * @param block The suspending block of code to execute.
+ * @return A [Resource] representing the outcome:
+ * - [Resource.Success] containing the data if the block succeeds.
+ * - [Resource.Error] containing the error message if an exception occurs.
+ * @throws CancellationException if the coroutine is canceled; this exception is rethrown
+ * to ensure proper coroutine lifecycle management.
+ */
+suspend fun <T> safeApiCall(block: suspend () -> T): Resource<T> {
+    return try {
+        Resource.Success(block())
+    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)